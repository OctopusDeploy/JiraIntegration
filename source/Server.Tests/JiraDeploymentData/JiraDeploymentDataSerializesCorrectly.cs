using System;
using Assent;
using Newtonsoft.Json;
using NUnit.Framework;
using Octopus.Server.Extensibility.JiraIntegration.Deployments;
using Octopus.Server.Extensibility.JiraIntegration.Environments;

namespace Octopus.Server.Extensibility.JiraIntegration.Tests.JiraDeploymentData
{
    [TestFixture]
    public class JiraDeploymentDataSerializesCorrectly
    {
        [Test]
        public void DataSerializesCorrectly()
        {
            var data = new OctopusJiraPayloadData
            {
                InstallationId = "12345",
                BaseHostUrl = "https://octopussample.com",
                DeploymentsInfo = new JiraPayloadData
                {
                    Deployments = new[]
                    {
                        new Extensibility.JiraIntegration.Deployments.JiraDeploymentData
                        {
                            DeploymentSequenceNumber = 11,
                            UpdateSequenceNumber = 3,
                            DisplayName = "Task Name",
                            Associations = new []
                            {
                                new JiraAssociation()
                                {
                                    AssociationType = JiraAssociationConstants.JiraAssociationTypeIssueKeys,
                                    Values = new [] { "JIR-1", "JIR-2"}
                                },
                            },
                            Url =
                                "https://octopussample.com/app#/Spaces-1/projects/foo/releases/1.0.0/deployments/deployments-123",
                            Description = "Task Description",
                            LastUpdated = new DateTimeOffset(new DateTime(2018, 10, 27)),
                            State = "in_progress",
                            Pipeline = new JiraDeploymentPipeline
                            {
                                Id = "Projects-234",
                                DisplayName = "Jira Project",
                                Url = "https://octopussample.com/app#/Spaces-1/projects/foo"
                            },
                            Environment = new JiraDeploymentEnvironment
                            {
                                Id = "env-123",
                                DisplayName = "Development",
                                Type = JiraEnvironmentType.development.ToString()
                            },
                            SchemeVersion = "1.0"
                        }
                    }
                }
            };
            var json = JsonConvert.SerializeObject(data, Formatting.Indented);
            this.Assent(json);
        }
    }
}