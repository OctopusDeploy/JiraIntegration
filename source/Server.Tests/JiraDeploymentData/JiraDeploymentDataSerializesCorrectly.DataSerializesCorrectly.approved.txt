{
  "InstallationId": "12345",
  "BaseHostUrl": "https://octopussample.com",
  "DeploymentsInfo": {
    "deployments": [
      {
        "deploymentSequenceNumber": 11,
        "updateSequenceNumber": 3,
        "displayName": "Task Name",
        "associations": [
          {
            "associationType": "issueKeys",
            "values": [
              "JIR-1",
              "JIR-2"
            ]
          }
        ],
        "url": "https://octopussample.com/app#/Spaces-1/projects/foo/releases/1.0.0/deployments/deployments-123",
        "description": "Task Description",
        "lastUpdated": "2018-10-27T00:00:00+10:00",
        "state": "in_progress",
        "pipeline": {
          "id": "Projects-234",
          "displayName": "Jira Project",
          "url": "https://octopussample.com/app#/Spaces-1/projects/foo"
        },
        "environment": {
          "id": "env-123",
          "displayName": "Development",
          "type": "development"
        },
        "id": "1.0"
      }
    ]
  }
}